mantidproject/mantid
# Rename and add fixes to CalculateResolution

***Note*** *The branch name and commits reference the wrong issue (#18842 instead of #18442)*

The following changes have been made to `CalculateResolution`:

- The algorithm has been renamed to `NRCalculateSlitResolution` as this algorithm is specific to neutron reflectometry, and the resolution it calculates is the slit resolution.
- The `TwoThetaLogName` property has been replaced by `ThetaLogName`. This still takes `Theta` as the default value, which was causing confusion before because it was being used as two theta. Is is now being used as theta, as the new property name suggests.
- Some errors in the resolution calculation have been fixed. The equation used now matches that in the spreadsheet in the linked issue.
- An error in the usage example has been fixed, where theta instead of two theta was being passed to the `TwoTheta` property.
- The output property `TwoThetaOut` has been removed because it is not used.

Fixes #18442 

**To test:**
Run this script:
```
ws = Load('INTER00013460')
theta = 0.7000204
print "%.6f" % (NRCalculateSlitResolution(Workspace = ws)*100)
print "%.6f" % (NRCalculateSlitResolution(Workspace = ws, TwoTheta=2*theta)*100)
print "%.6f" % (NRCalculateSlitResolution(Workspace = ws, ThetaLogName="Theta")*100)
```
You should get the same result for each call to `NRCalculateSlitResolution`:
```
3.402745
3.402745
3.402745
```

**Release Notes** 
*Added to the release notes*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
